### DISCUSSION TITLE
UI and layout design

### DISCUSSION THREAD

**Discussion by rutipsi** [sayaka-sh/spmp#82] [Ideas]
Current state of the UI is okay. 
Lyrics on top is great! I love that and never want it to change, seems very unique. 
Search bar or icon (that can drop down search bar) can be moved in between top lyrics bar and music category i.e. workout - relax - ...
or maybe below music category i.e. workout - relax - ..
![Screenshot 2023-07-19 164848](https://github.com/toasterofbread/spmp/assets/96968562/35385cd7-1ebb-4f80-bae3-89a23a9dad56)

Library, playlists, liked songs and settings can be moved to account section. 
Full screen lyrics with dedicated button would be great. Like button can be moved to the line of play/pause button. 
I guess shuffle/repeat button is in work. 
I love the overall feel and UI.

**Comment by sayaka-sh** [comment]
>  Lyrics on top is great! I love that and never want it to change, seems very unique.

Thanks, I'm really glad I had the idea for that. Can't believe other apps don't do that.

>  Search bar or icon (that can drop down search bar) can be moved in between top lyrics bar and music category i.e. workout - relax - ...

How about a search icon next to the category bar? I'd rather not take up more vertical space there. Like this: 
![search-concept](https://github.com/toasterofbread/spmp/assets/44363049/fe146309-2b75-4525-b91f-44685a9a739e)

> Library, playlists, liked songs and settings can be moved to account section.

Could you explain a bit more? I'm not sure what you mean by this

>  Full screen lyrics with dedicated button would be great

There's already a larger lyrics screen if you click on the large song image and click the music note button. Do you mean a larger version of that?
You can also access it quickly by long pressing on the top bar lyrics.

<img src="https://github.com/toasterofbread/spmp/assets/44363049/ef447d21-6393-4d81-a06a-f392cc604a5e" height=500></img>

>  I guess shuffle/repeat button is in work.

What do you mean? They should both be working.

> Like button can be moved to the line of play/pause button.

I'll consider that. The area around the play/pause and skip buttons definitely isn't final.

**Comment by rutipsi** [comment]
It's good, other apps don't do that. Btw [Salt Player](https://github.com/Moriafly/SaltPlayerSource/) (offline player) does something like this. Good use of empty space in first pic. 
 
<img src="https://github.com/toasterofbread/spmp/assets/96968562/50f0ac1c-79f4-47f7-890d-ff083f1da3e2" height=500/>

Yeah, good idea. Looking minimalistic and nice. 
![254628046-fe146309-2b75-4525-b91f-44685a9a739e](https://github.com/toasterofbread/spmp/assets/96968562/47a0cc57-2af7-46ae-bb94-3562821ffe86)

My idea was to get rid of these buttons. It's nice but it could be easier to access. Search icon could be at top, as you shown. 
![Screen33112](https://github.com/toasterofbread/spmp/assets/96968562/64102984-17ff-41ce-b36b-a3fd094fc80b)

These two sections, can be can be moved here. Naturally people will go there to access settings. 

<img src="https://github.com/toasterofbread/spmp/assets/96968562/57b07968-583a-43bf-bb88-e36446ba698d" height=500/>

This thing is really good for music discovery. Library can have dedicated button like this somewhere on front page (cause people would be accessing library more often) or like I shown in above image. 
![](https://github.com/toasterofbread/spmp/assets/96968562/4c850994-a587-4a21-9767-cfc3ecc89dd1)

By full screen lyrics, I meant something like this. People who love synced lyrics, this would be a nice experience for them. 
With a dedicated lyrics beside play/pause or somewhere near that.  
<img src="https://github.com/toasterofbread/spmp/assets/96968562/12925861-bd83-49f7-be19-5fb2d04bb66a" height=500/>

**Comment by sayaka-sh** [comment]
> Yeah, good idea. Looking minimalistic and nice.
> ![](https://user-images.githubusercontent.com/96968562/254674260-4c850994-a587-4a21-9767-cfc3ecc89dd1.png)

Cool, ll get that implemented at some point.

<br>

> My idea was to get rid of these buttons. It's nice but it could be easier to access. 

I agree, and I like the idea of merging the pages into the profile. 

Since the profile can only be accessed while logged in, what I'll do is add a new UI page accessed from the top right of the main page (where the profile button is now) which will show a small overview of your profile at the top, and contain what's currently in the library page below that. I'll include a button to open the settings somewhere in there.

<br>

> By full screen lyrics, I meant something like this. People who love synced lyrics, this would be a nice experience for them.
With a dedicated lyrics beside play/pause or somewhere near that.

I was already thinking about a fullscreen lyrics page so that's great. I'll also add a dedicated button to the main player, good idea. The space there is anything but final so I'm open to more suggestions for what to add there.

**Reply by rutipsi** [reply]
Good to hear, that you were thinking the same and you like it. I'll suggest as I get some ideas.

**Reply by sayaka-sh** [reply]
[Moved the search page button](https://github.com/toasterofbread/spmp/commit/3ccb48078e77f8d447b9b64b962887736603cbbb) like I described, the new page for the library and settings should follow soon.

**Comment by Zaxx69** [comment]
![Screenshot_2023-07-23-21-12-30-69_e82a23cb50dd4a2f2e01332c4ab3b632.jpg](https://github.com/toasterofbread/spmp/assets/138523882/4dd2eeec-5dda-42e6-90bc-2f7eeeba42ea)

Bro the app is great, i mean it's better than other apps. The thing i wanted to share is that the cross box blocks some of the settings. I don't know coding but i have theoritical solution, let the cross button appear only when we move the screen up and make it disappear when we move down. I hope it helps

**Reply by Zaxx69** [reply]
Another thing i wanna talk about is that whenever i open the app and click on the top tabs like comute, energise, etc i can't go back to the main page even if i click back on my phone and brother can you add an feature to add our own font please

**Reply by sayaka-sh** [reply]
> The thing i wanted to share is that the cross box blocks some of the settings

Yeah, the page is supposed to have extra padding at the bottom for the buttons to fit into. I think I removed it recently and forgot to put it back, will fix soon. 

> Another thing i wanna talk about is that whenever i open the app and click on the top tabs like comute, energise, etc i can't go back to the main page even if i click back on my phone

You can disable the filter by clicking on it again.

**Reply by sayaka-sh** [reply]
The padding I mentioned should be [fixed](https://github.com/toasterofbread/spmp/commit/3bddeabdc301865628facd750085886f2a7e2a7b).

**Reply by Zaxx69** [reply]
@toasterofbread 
There are other bugs too like
1. Even if i clear the app from my recent app, the song still plays and i can't stop it (it happens sometimes)
2. In the playlist section, even if i arrange the songs according to their, they do not arrange and they revert back to original playlist whenever i leave the app
3. Bro can you add a home button beside the search button, it can make navigation easy
4. Also in the playlist buttons 
![IMG_20230724_102825.jpg](https://github.com/toasterofbread/spmp/assets/138523882/f73e4a24-8257-45c7-a0f9-b072ec9b89a7)

1) plus button (red circle) - doesn't work and shows error

2) the four slash button (purple circle) - doesn't work and crashes the app

3) the edit button (white circle) - it reverts back my playlist name and picture whenever i leave the app and shows the default as you can see above

4)the three lines (yellow circle) - as mentioned in number 2

These are some bugs i found, can you fix them brother and this app will be great

**Reply by Zaxx69** [reply]
Bro please reply

**Reply by sayaka-sh** [reply]
Thanks for the reports, but please put them into issue(s) from now on and include more information. This thread is for discussing changes to the UI design rather than the implementation itself.

> Even if i clear the app from my recent app, the song still plays and i can't stop it (it happens sometimes)

Playback continuing after the app is closed is the intended behaviour (although I'll probably add an option for it at some point). When you say you can't stop it, what do you mean exactly? Does pressing pause do nothing? Or do the buttons on the notification not display at all? (this is a known bug on some older Android versions)

> In the playlist section, even if i arrange the songs according to their, they do not arrange and they revert back to original playlist whenever i leave the app

What do you mean by arrange? Assuming you mean the playlist sort order (since you mention below that the reorder button with the purple circle crashes the app), that's intended behaviour. The sort is just for the current UI state, whereas reordering is supposed to be permanent.

> Bro can you add a home button beside the search button, it can make navigation easy

Which search button? I'm guessing you mean the button would lead to the main app page rather than the system home page.

> plus button (red circle) - doesn't work and shows error

Not implemented yet

> the four slash button (purple circle) - doesn't work and crashes the app

Is there an error message displayed? If not, does the app freeze and then crash or just crash immediately?

> the edit button (white circle) - it reverts back my playlist name and picture whenever i leave the app and shows the default as you can see above

Strange. Does it only revert when you reopen the app, or also when you close the playlist and open it again? Is it a local playlist or did you upload it to your account?

**Reply by Zaxx69** [reply]
@toasterofbread
1. I meant that even if i CLOSE my app, and CLEAR it from recent apps(every phone has that where you can see every app used). The music notification still persist on my notification bar and i am on android 13

2. Yeah i am talking about sorting, but you misunderstood, let me get this clear
Let's say i sort my songs by name, the following problem arises -
. Not all the songs are sorted by the catogory
. When i leave the playlist section after sorting, it reverts back
Now about forced arrange, i tried to do it but the cross button took some space so i couldn't click on the button as it was blocked but i tried and a song got overlapped over another song and stopped working. I closed the app and restarted it.

3. yeah, i think it will be better

4. Ok,

5. Yeah i told you the app froze and stoped working so i restarted the app again

6. No it reverts back when i close the app
It is a local playlist

7. I will tell you the latest bug in the issue section, i am telling these, because the app is good and with this bugs removed and with implementing new features, it will be great

**Reply by sayaka-sh** [reply]
> I meant that even if i CLOSE my app, and CLEAR it from recent apps

Yes, as I said that's the intended behaviour.

>  Not all the songs are sorted by the catogory

Which ones are and which ones aren't? Is it consistent, and do the other sort options work? A screenshot would be useful.

> When i leave the playlist section after sorting, it reverts back

This is also the intended behaviour, although I'd be able to add an option to change this.

> Now about forced arrange

By forced arrange, do you mean the purple circled button? I though that caused a crash for you. 

> i tried to do it but the cross button took some space 

Please be more specific, which cross button?

> so i couldn't click on the button as it was blocked 

Again, which button is `the button`?

> but i tried and a song got overlapped over another song and stopped working

I'm sorry but I have no idea what you're talking about here 😓

> yeah, i think it will be better

Again, which search button?

**Comment by Zaxx69** [comment]
Some ui changes opinion :-
1. Amoled display support - the entire background will be black and only other elements with be highlighted. example
![Screenshot_2023-07-25-11-33-54-95_a5af3585a93e2016f6667aaf7e1e2f32.jpg](https://github.com/toasterofbread/spmp/assets/138523882/1dbe5ed0-e42e-4e2b-bb4f-dd2788768f8a)

2. Font change ( i know someone reported this issue and i know you said you will fix it future) 

3. Can you make this button appear
 ![Screenshot_2023-07-25-11-36-37-97_e82a23cb50dd4a2f2e01332c4ab3b632.jpg](https://github.com/toasterofbread/spmp/assets/138523882/ed172639-9db3-4e23-a42f-b3ac7200aabf)

only when we are scrolling up and not down, example a blackhole (foss music app) had this problem but they fixed it like this. 

4. A search button for searching artist in the artist section 
![Screenshot_2023-07-25-11-39-02-32_e82a23cb50dd4a2f2e01332c4ab3b632.jpg](https://github.com/toasterofbread/spmp/assets/138523882/59832741-4297-4d27-9279-714258795791)

5. If we could upload our own image to the playlist image 
![Screenshot_2023-07-25-11-40-31-41_e82a23cb50dd4a2f2e01332c4ab3b632.jpg](https://github.com/toasterofbread/spmp/assets/138523882/e40a7010-124a-4cf7-9ced-16374efc1c2d)

6. You can use the plus button in the playlist section for the following thing 
. To add songs from recently heard,liked songs, other playlist, downloaded songs 
example spotify has this feature 

7. Thank you for your hard work, we are the users, we will detect the problem and also help solving it. We are with you.

**Reply by sayaka-sh** [reply]
> Amoled display support - the entire background will be black and only other elements with be highlighted. example

Just edit a theme and set the background to black

> Font change

Like you said, this is already planned. I already have a font in mind, but I'm open to suggestions (leave them in #69).

> Can you make this button appear

That button has/will be removed (this was discussed earlier in the thread). I'm pretty strongly against making things appear or disappear based on your scrolling direction, but naturally I can always add an option to enable/disable behaviour. Again though, that button has been removed and I'm probably going to be removing similar buttons from other pages as well.

> A search button for searching artist in the artist section

Good idea, I'll add this.

> If we could upload our own image to the playlist image

I suppose I could add that, but are you sure it'd be that useful? It won't apply to the playlist on your YouTube account since YouTube doesn't implement that functionality and you can already pick the thumbnail of a song in the playlist.

> You can use the plus button in the playlist section for the following thing

That's more or less the plan, yeah.

> Thank you for your hard work, we are the users, we will detect the problem and also help solving it

Thanks, I really appreciate the suggestions.

**Comment by sayaka-sh** [comment]
Thread is getting too long, I made a [UI discussion category](https://github.com/toasterofbread/spmp/discussions/categories/ui-discussion) to be used instead.
